api/src/main/java/org/openmrs/module/fhir/api/helper/FHIRClientHelper.java

```diff
@@ -89,6 +100,28 @@ public Class resolveCategoryByCategory(String category) {
 				{ new FHIRHttpMessageConverter(), new StringHttpMessageConverter() });
 	}
 
+	@Override
+	public boolean compareResourceObjects(String category, Object from, Object dest) {
+		boolean result = false;
+		switch (category) {
+			case CATEGORY_PATIENT:
+				result = FHIRPatientUtil.compareCurrentPatients(dest, from);
+				break;
+			case CATEGORY_ENCOUNTER:
+				result = FHIREncounterUtil.compareCurrentEncounters(dest, from);
+				break;
+			case CATEGORY_VISIT:
+				result = FHIREncounterUtil.compareCurrentEncounters(dest, from);
```

Comment:
WIll a visit e always related to one encouter?

Reply:
The OpenMRS Visit is mapped to the FHIR Encounter. This code is moved from:
https://github.com/openmrs/openmrs-module-sync2/blob/a4b032c623c89bd6f39e6373649731e74733f243/api/src/main/java/org/openmrs/module/sync2/api/utils/SyncUtils.java#L186